egs2/librispeech/discrete_asr1/local/data.sh

```diff
@@ -0,0 +1,452 @@
+        else
+            echo "Unsupported hubert version ${ckpt_version}" && exit 1;
+        fi
+        # shellcheck disable=SC2206
+        ckpt_name=$(echo ${model_url} | awk -F/ '{print $NF}')
+    elif [ ${ckpt_type} == "wavlm" ]; then
+        # WavLM
+        if [ ! -d "local/wavlm" ]; then
+            echo "Download wavlm model packages"
+            mkdir -p "local/wavlm"
+            wget -O "local/wavlm/WavLM.py" "https://raw.githubusercontent.com/microsoft/unilm/master/wavlm/WavLM.py"
+            wget -O "local/wavlm/modules.py" "https://raw.githubusercontent.com/microsoft/unilm/master/wavlm/modules.py"
+        else
+            echo "local/wavlm already exists. Skip downloading"
+        fi
+
+        if [ ${ckpt_version} == "base" ]; then
+            # WavLM_base
+            model_url="https://valle.blob.core.windows.net/share/wavlm/WavLM-Base.pt?sv=2020-08-04&st=2023-03-01T07%3A51%3A05Z&se=2033-03-02T07%3A51%3A00Z&sr=c&sp=rl&sig=QJXmSJG9DbMKf48UDIU1MfzIro8HQOf3sqlNXiflY1I%3D"
+        elif [ ${ckpt_version} == "large" ]; then
+            # WavLM_large
+            model_url="https://valle.blob.core.windows.net/share/wavlm/WavLM-Large.pt?sv=2020-08-04&st=2023-03-01T07%3A51%3A05Z&se=2033-03-02T07%3A51%3A00Z&sr=c&sp=rl&sig=QJXmSJG9DbMKf48UDIU1MfzIro8HQOf3sqlNXiflY1I%3D"
+        else
+            echo "Unsupported wavlm version ${ckpt_version}" && exit 1;
+        fi
+        ckpt_name="wavlm_${ckpt_version}.pt"
+    else
+        echo "Unsupported ssl type ${ckpt_type}" && exit 1;
+    fi
+}
+
+feats_dir=${feats_root}/${model_type}_${model_version}/l${layer}
+if [ ${stage} -le 3 ] && [ ${stop_stage} -ge 3 ] && ! [[ " ${skip_stages} " =~ [[:space:]]3[[:space:]] ]]; then
+    log "Stage 3: dump features"
+
+    # download ssl ckpt
+    mkdir -p "./ckpt"
+
+    download_model_ckpts "${model_type}" "${model_version}"
+    if [ ! -f "./ckpt/${ckpt_name}" ]; then
+        wget -O "./ckpt/${ckpt_name}" ${model_url}
+    fi
+
+    for dset in ${test_sets} "${train_set}" ; do
+        log "Dump SSL ${dset} features to ${feats_dir}/${dset}"
+
+        # 1. Split the key file
+        utils/copy_data_dir.sh --validate_opts --non-print data/dump_audio/"${dset}" "${feats_dir}/${dset}"
+        rm -f "${feats_dir}/${dset}"/{segments,wav.scp,reco2file_and_channel,reco2dur}
+
+        output_dir="${feats_dir}/${dset}/data"
+        mkdir -p "${output_dir}"
+        _logdir="${feats_dir}/${dset}/logdir"
+        mkdir -p "${_logdir}"
+
+        nutt=$(<"data/dump_audio/${dset}"/wav.scp wc -l)
+        _nj=$((gpu_nj<nutt?gpu_nj:nutt))
+
+        key_file="data/dump_audio/${dset}"/wav.scp
+        split_scps=""
+        for n in $(seq ${_nj}); do
+            split_scps+=" ${_logdir}/wav.${n}.scp"
+        done
+        # shellcheck disable=SC2086
+        utils/split_scp.pl "${key_file}" ${split_scps}
+
+        ${cuda_cmd} --gpu 1 JOB=1:${_nj} ${_logdir}/dump_features.JOB.log \
+            ${python} local/dump_hubert_or_wavlm_feature.py \
```

Comment:
why not using the s3prl for ssl features, I believe that could be easier to maintain?

Reply:
I added s3prl support for this task, although I'm not sure if it matches the performance. 
But it is better to keep the script in ssl1 for better support of espnet-hubert models.